packages/material-ui/src/InputBase/Textarea.test.js

```diff
@@ -1,164 +1,191 @@
-    it('should be call the callback', () => {
+    it('should update when uncontrolled', () => {
       const handleChange = spy();
-      const wrapper = shallow(<TextareaNaked classes={{}} value="x" onChange={handleChange} />);
-      assert.strictEqual(wrapper.find('textarea').length, 3);
-
-      const refs = assignRefs(wrapper);
-      const event = { target: { value: 'xx' } };
-      refs.inputRef.simulate('change', event);
-      assert.strictEqual(wrapper.instance().value, 'xx');
+      const wrapper = mount(<Textarea onChange={handleChange} />);
+      setLayout(
+        wrapper,
+        {
+          'box-sizing': 'content-box',
+        },
+        30,
+        15,
+      );
+      assert.strictEqual(getHeight(wrapper), 0);
+      wrapper
+        .find('textarea')
+        .at(0)
+        .simulate('change');
+      wrapper.update();
+      assert.strictEqual(getHeight(wrapper), 30);
       assert.strictEqual(handleChange.callCount, 1);
-      assert.deepEqual(handleChange.args[0], [event]);
     });
-  });
 
-  describe('resize', () => {
-    let clock;
+    it('should handle border-box', () => {
+      const border = 5;
+      const wrapper = mount(<Textarea />);
+      setLayout(
+        wrapper,
+        {
+          'box-sizing': 'border-box',
+          'border-bottom-width': `${border}px`,
+        },
+        30,
+        15,
+      );
+      wrapper.setProps({});
+      wrapper.update();
+      assert.strictEqual(getHeight(wrapper), 30 + border);
+    });
 
-    before(() => {
-      clock = useFakeTimers();
+    it('should handle content-box', () => {
+      const padding = 5;
+      const wrapper = mount(<Textarea />);
+      setLayout(
+        wrapper,
+        {
+          'box-sizing': 'content-box',
+          'padding-top': `${padding}px`,
+        },
+        30,
+        15,
+      );
+      wrapper.setProps({});
+      wrapper.update();
+      assert.strictEqual(getHeight(wrapper), 30 - padding);
     });
 
-    after(() => {
-      clock.restore();
+    it('should handle rowsMin', () => {
+      const rowsMin = 3;
+      const lineHeight = 15;
+      const wrapper = mount(<Textarea rowsMin={rowsMin} />);
+      setLayout(
+        wrapper,
+        {
+          'box-sizing': 'content-box',
+        },
+        30,
+        lineHeight,
+      );
+      wrapper.setProps({});
+      wrapper.update();
+      assert.strictEqual(getHeight(wrapper), lineHeight * rowsMin);
     });
 
-    it('should handle the resize event', () => {
-      const wrapper = shallow(<TextareaNaked classes={{}} />);
-      const refs = assignRefs(wrapper);
-      refs.textareaShadowRef.scrollHeight = 43;
-      refs.singlelineShadowRef.scrollHeight = 43;
-      wrapper
-        .find('EventListener')
-        .at(0)
-        .simulate('resize');
-      assert.strictEqual(wrapper.state().height, 19);
-      clock.tick(166);
-      assert.strictEqual(wrapper.state().height, 43);
+    it('should handle rowsMax', () => {
```

Comment:
Same as above. In my experience these generic descriptions are almost always a code smell. Either they indicate testing of implementation details or don't help understand how the component should behave. It looks like these tests are derived from this specific implementation and not the other way around.